server/controllers/automationController.js

```diff
@@ -178,15 +178,16 @@ function getPaginationMeta(page, count, limit) {
  */
 const paginationData = async (req, res) => {
   const orderBy = order.map(item => item.join(' ')).join();
-  const limit = parseInt(req.query.limit, 10) || 10;
+  let limit = parseInt(req.query.limit, 10) || 10;
+  if (limit === -1) {
+    limit = (await automation.findAll()).length;
```

Comment:
This is quite inefficient as you make the query to get all automations only to get a count/length, then another call will be made to get the same automations again.

Reply:
Noted, will refactor